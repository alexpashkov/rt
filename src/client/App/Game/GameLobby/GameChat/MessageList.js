import React from 'react';
import PropTypes from 'prop-types';

const MessageList = ({ messages }) => (
  <div>
    {messages.length
      ? messages.map((message, i) => (
          <pre key={i}>{JSON.stringify(message)}</pre>
        ))
      : 'Be the first who will send a message!'}
  </div>
);

MessageList.propTypes = {
  messages: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      login: PropTypes.string.isRequired,
      message: PropTypes.string.isRequired
    })
  )
};

export default MessageList;
