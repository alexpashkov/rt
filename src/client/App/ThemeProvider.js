import React from 'react';
import { ThemeProvider as EmotionThemeProvider } from 'emotion-theming';
import theme from '../theme';

export const ThemeProvider = ({ children }) => (
  <EmotionThemeProvider theme={theme}>{children}</EmotionThemeProvider>
);

export default ThemeProvider;
