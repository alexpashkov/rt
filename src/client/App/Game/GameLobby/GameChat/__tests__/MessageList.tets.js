import React from "react";
import MessageList from '../MessageList';
import renderer from 'react-test-renderer';
import ThemeProvider from "../../../../ThemeProvider"

describe('GameChat', () => {
  it('Renders without throwing an exception', () => {
    const component = renderer.create(<ThemeProvider><MessageList /></ThemeProvider>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
