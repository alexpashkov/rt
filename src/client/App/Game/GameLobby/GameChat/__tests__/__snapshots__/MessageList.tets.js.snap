// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameChat Renders without throwing an exception 1`] = `
<div
  className="css-cre6xo euynx711"
/>
`;
